### ISSUE TITLE
issue regarding metrics

### ISSUE THREAD

**Issue by kocemir** [OmicsML/CellPLM#14]
Hey, I want to ask about the metrics that you used. you are using torch.metrics.functional.classification. but if I switch to sklearn, results differ significantly. Here is the code that I wrote. do you have any guidance regarding the reason?

    **acc= accuracy_score(pred_labels.detach().cpu().numpy(),true_labels.detach().cpu().numpy())
    f1= f1_score(pred_labels.detach().cpu().numpy(),true_labels.detach().cpu().numpy(),average="macro")
    precision= precision_score(pred_labels.detach().cpu().numpy(),true_labels.detach().cpu().numpy(),average="macro")
    recall= recall_score(pred_labels.detach().cpu().numpy(),true_labels.cpu().detach().numpy(),average="macro")**

----
Edit: I understood the reason. Since multiclass_accuracy function of the torchmetrics assumes "average=macro" by default. Why do you choose it, in particular for the accuracy, is it better to use average=macro in this case, or weighted for all of the metrics.

**Comment by wehos** [comment]
Macro F1 is the golden standard and other metrics are just axiliary. Regarding your question about accuracy, macro average makes sense in multi-class classification with unbalanced labels, like our cases. Feel free to let me know if you have further questions.

**Comment by kocemir** [comment]
Thank you very much for your answer.  I tried to reproduce the results in your work, got almost exactly same macro F1 for two datasets. Really insightfull work !
